app/helpers/samples_helper.rb

```diff
@@ -60,14 +61,14 @@ def generate_sample_list_csv(formatted_samples)
                         collection_location: collection_location,
                         host_genome: derived_output && derived_output[:host_genome_name] ? derived_output[:host_genome_name] : '',
                         notes: db_sample && db_sample[:sample_notes] ? db_sample[:sample_notes] : '',
-                        insert_size_median: pipeline_run ? pipeline_run.dig(:insert_size_metric_set, :median) : '',
-                        insert_size_mode: pipeline_run ? pipeline_run.dig(:insert_size_metric_set, :mode) : '',
-                        insert_size_median_absolute_deviation: pipeline_run ? pipeline_run.dig(:insert_size_metric_set, :median_absolute_deviation) : '',
-                        insert_size_min: pipeline_run ? pipeline_run.dig(:insert_size_metric_set, :min) : '',
-                        insert_size_max: pipeline_run ? pipeline_run.dig(:insert_size_metric_set, :max) : '',
-                        insert_size_mean: pipeline_run ? pipeline_run.dig(:insert_size_metric_set, :mean) : '',
-                        insert_size_standard_deviation: pipeline_run ? pipeline_run.dig(:insert_size_metric_set, :standard_deviation) : '',
-                        insert_size_read_pairs: pipeline_run ? pipeline_run.dig(:insert_size_metric_set, :read_pairs) : '', }
+                        insert_size_median: pipeline_run && pipeline_run.insert_size_metric_set && pipeline_run.insert_size_metric_set.median ? pipeline_run.insert_size_metric_set.median : '',
```

Comment:
I think the `dig` method would shorten these lines but no worries if re-testing is annoying

See https://tiagoamaro.com.br/2016/08/27/ruby-2-3-dig/.

Reply:
The dig method caused the issue, was I using it incorrectly? I think it is only for hashes and not class members.

Reply:
nevermind! I didn't even read the red lines.